Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,326 @@
+    $EDiscoveryEndpoint = Get-IntraOrganizationConfiguration -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Select-Object OnPremiseDiscoveryEndpoint
+    $SPDomainsOnprem = Get-SharingPolicy -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Format-List Domains
+    $SPOnprem = Get-SharingPolicy  -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Select-Object *
+
+    if ($Org -contains 'ExchangeOnPremise' -or -not $Org) {
+        #region DAuth Checks
+        if ($Auth -like "DAuth" -OR -not $Auth -or $Auth -like "All") {
+            Write-Host $TestingDAuthConfiguration
+            OrgRelCheck -OrgRelParameter $OrgRel
+            PrintDynamicWidthLine
+            FedInfoCheck
+            FedTrustCheck
+            AutoDVirtualDCheck
+            PrintDynamicWidthLine
+            EWSVirtualDirectoryCheck
+            AvailabilityAddressSpaceCheck
+            TestFedTrust
+            TestOrgRel
+        }
+        #endregion
+        #region OAuth Check
+        if ($Auth -like "OAuth" -or -not $Auth -or $Auth -like "All") {
+            Write-Host $TestingOAuthConfiguration
+            # PrintDynamicWidthLine
+            IntraOrgConCheck
+            PrintDynamicWidthLine
+            AuthServerCheck
+            PrintDynamicWidthLine
+            PartnerApplicationCheck
+            PrintDynamicWidthLine
+            ApplicationAccountCheck
+            PrintDynamicWidthLine
+            ManagementRoleAssignmentCheck
+            PrintDynamicWidthLine
+            AuthConfigCheck
+            PrintDynamicWidthLine
+            CurrentCertificateThumbprintCheck
+            PrintDynamicWidthLine
+            AutoDVirtualDCheckOAuth
+            $AutoDiscoveryVirtualDirectoryOAuth
+            PrintDynamicWidthLine
+            EWSVirtualDirectoryCheckOAuth
+            PrintDynamicWidthLine
+            AvailabilityAddressSpaceCheckOAuth
+            PrintDynamicWidthLine
+            OAuthConnectivityCheck
+            PrintDynamicWidthLine
+        }
+        #endregion
+    }
+    # EXO Part
+    if ($Org -contains 'ExchangeOnline' -OR -not $Org) {
+        #region ConnectExo
+
+        Write-Host -ForegroundColor Green $CollectingExoAvailabilityInformation
+        # Check if the ExchangeOnlineManagement module is already installed
+        if (-not (Get-Module -Name ExchangeOnlineManagement -ListAvailable)) {
+            # If not installed, then install the module
+            Write-Host -ForegroundColor Yellow $ExchangeOnlinePowershellModuleMessage
+            Install-Module -Name ExchangeOnlineManagement -Force
+            PrintDynamicWidthLine
+        } else {
+            Write-Host $ExchangeOnlineModuleAvailableMessage
+            $ExoModuleVersion = Get-Module -Name ExchangeOnlineManagement -ListAvailable | Format-List name, Version
+            $ExoModuleVersion
+            PrintDynamicWidthLine
+        }
+
+        Connect-ExchangeOnline -ShowBanner:$false
```

Comment:
Do not connect, this should already be done by the admin prior to running the script. Only thing we should be doing is verifying that we are connected.

Reply:
Removing